[macOS] Declare notification names observed by WebKit in NSWindowSPI.h

aestes opened the pull request:
#### 890b606d3b29fa00a87b9b92645df4e2c5c836de
<pre>
[macOS] Declare notification names observed by WebKit in NSWindowSPI.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=256184">https://bugs.webkit.org/show_bug.cgi?id=256184</a>

Reviewed by Wenson Hsieh.

Ensured `NSWindowDidOrderOffScreenNotification`, `NSWindowDidOrderOnScreenNotification`, and
`_NSWindowDidChangeContentsHostedInLayerSurfaceNotification` are declared in NSWindowSPI.h then
used those constants instead of string literals in WebViewData.m and WebViewImpl.m. Note that
`NSWindowDidOrderOnScreenNotification` is defined to `@&quot;_NSWindowDidBecomeVisible&quot;` in AppKit.

* Source/WebCore/PAL/pal/spi/mac/NSWindowSPI.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKWindowVisibilityObserver startObserving:]):
(-[WKWindowVisibilityObserver stopObserving:]):
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
(-[WebWindowVisibilityObserver startObserving:]):
(-[WebWindowVisibilityObserver stopObserving:]):

Canonical link: <a href="https://commits.webkit.org/263592@main">https://commits.webkit.org/263592@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/92aa0f2f2fba90bd2b74f4580ce46112ac9dcc3e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5095 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5226 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5403 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6616 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5178 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5089 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5426 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5202 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5426 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5182 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5268 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4572 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6632 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2755 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4570 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9841 "2 flakes 173 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4618 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4643 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6255 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5060 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4153 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4543 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1236 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8623 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4908 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -306,8 +306,8 @@ - (void)startObserving:(NSWindow *)window
-    [defaultNotificationCenter addObserver:self selector:@selector(_windowDidOrderOffScreen:) name:@"NSWindowDidOrderOffScreenNotification" object:window];
-    [defaultNotificationCenter addObserver:self selector:@selector(_windowDidOrderOnScreen:) name:@"_NSWindowDidBecomeVisible" object:window];
+    [defaultNotificationCenter addObserver:self selector:@selector(_windowDidOrderOffScreen:) name:NSWindowDidOrderOffScreenNotification object:window];
+    [defaultNotificationCenter addObserver:self selector:@selector(_windowDidOrderOnScreen:) name:_NSWindowDidBecomeVisible object:window];
```
Do we need `ALLOW_DEPRECATED_DECLARATIONS_BEGIN`/`END` around usages of `_NSWindowDidBecomeVisible` in this file? (It looks like this patch guards `_NSWindowDidBecomeVisible` behind that in WebKitLegacy, but not in WebKit2)

aestes replied:
Oops, yes. Not sure how I missed that.

aestes replied:
Even better, we can just use the non-deprecated `NSWindowDidOrderOnScreenNotification`

aestes committed:
[macOS] Declare notification names observed by WebKit in NSWindowSPI.h
https://bugs.webkit.org/show_bug.cgi?id=256184

Reviewed by Wenson Hsieh.

Ensured `NSWindowDidOrderOffScreenNotification`, `NSWindowDidOrderOnScreenNotification`, and
`_NSWindowDidChangeContentsHostedInLayerSurfaceNotification` are declared in NSWindowSPI.h then
used those constants instead of string literals in WebViewData.m and WebViewImpl.m. Note that
`NSWindowDidOrderOnScreenNotification` is defined to `@"_NSWindowDidBecomeVisible"` in AppKit.

* Source/WebCore/PAL/pal/spi/mac/NSWindowSPI.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKWindowVisibilityObserver startObserving:]):
(-[WKWindowVisibilityObserver stopObserving:]):
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
(-[WebWindowVisibilityObserver startObserving:]):
(-[WebWindowVisibilityObserver stopObserving:]):

Canonical link: https://commits.webkit.org/263592@main
